reinstate miatoll

phil-hands opened the pull request:
This attempts to reinstate miatoll support to the installer.

Marked as WiP because I note that it currently contains no mention of boot_a/b,  dtbo_a/b or vbmeta_a/b,
all of which are mentioned in `flash_all.sh`, so presumably are important.

I have run this against the my RedMi Note 9S, and it did things, looked correct, and when the phone rebooted, it worked, but I suspect that is in part because I'd previously installed the same phone with `flash_all.sh`, which had presumably done things with the AB partitioning, which this yml file then didn't mess up.

I'm happy to restore the stock firmware in order to test that it does a proper job, but first the AB stuff would presumably need to be fixed in the YAML.

Also, I've no idea about cutie -- I guess that bit of the config ought to be removed until someone has chance to test it.

phil-hands committed:
recover miatoll.yml, removed in d0dd848

phil-hands committed:
kludge miatoll to use recent image URLs

g7 commented:
Thanks for the PR! Added @bellegarde-c to the reviewers as he actually has the device, but looks good to me at first glance.

I think this snippet can be added before the partition flashing (`fastboot:flash`):
```
          - fastboot:set_active:
              slot: "a"
```

that should force the first slot. Then to be sure the `boot`, `dtbo` and `vbmeta` partitions can be changed to `boot_a`, `dtbo_a` and `vbmeta_a` so that they're forcibly flashed there.

During normal dist-upgrades, new kernels would automatically be flashed to the other slot (boot_b, dtbo_b, vbmeta_b), so even if we don't flash it during initial installation is not a problem (we just need to be sure that the selected slot is the correct one).

> Also, I've no idea about cutie -- I guess that bit of the config ought to be removed until someone has chance to test it.

Agreed.

phil-hands committed:
miatoll: explicitly set & flash slot a

as suggested by g7:

  https://github.com/droidian-devices/installer-configs/pull/2#issuecomment-2818068836

phil-hands committed:
miatoll: remove cutie references

These would need to be tested by someone that knows what cutie is. Until
that happens, offering it as an option gives a false impression of
usefulness.

phil-hands commented:
I've now applied what I think @g7 [suggested](https://github.com/droidian-devices/installer-configs/pull/2#issuecomment-2818068836), but I've not fully tested the result.

I ran the installer, and it no longer offers me cutie options, but I've not yet tried using it to flash my phone.  Given that it worked (on my already flash_all.sh installed phone) without the change, it's not clear to me that it succeeding with these changes would actually prove very much.

bellegarde-c reviewed (approved):
Looks OK for me.

g7 commented:
Thanks, I think we can merge it (once the WiP tag is removed)

phil-hands commented:
Since people seem happy with it, I got rid of the WiP -- hopefully that's the correct thing to have done.

g7 commented:
Thank you!
